src/main/java/uk/gov/hmcts/reform/sscs/service/ccdupdate/CcdLocationUpdateService.java

```diff
@@ -0,0 +1,109 @@
+package uk.gov.hmcts.reform.sscs.service.ccdupdate;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.exception.UpdateCaseException;
+import uk.gov.hmcts.reform.sscs.model.VenueDetails;
+import uk.gov.hmcts.reform.sscs.model.messaging.HmcMessage;
+import uk.gov.hmcts.reform.sscs.service.venue.VenueRpcDetailsService;
+
+import java.util.ArrayList;
+import java.util.List;
+import javax.validation.Valid;
+
+import static java.util.Objects.isNull;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CcdLocationUpdateService {
+
+    public static final String TEMPLATE_UPDATE_VENUE_ERROR = "Failed to update venue for Case Id %s:%n";
+
+    private final VenueRpcDetailsService venueRpcDetailsService;
+
+    public void updateVenue(HmcMessage hmcMessage, @Valid SscsCaseData sscsCaseData) throws UpdateCaseException {
```

Comment:
Don't like that there is so much happening in this method and by extension class.. the error handling should be included in the helper methods themselves and these helper methods should be extracted to a helper class that is used by this class.. separation of concerns.